### ISSUE TITLE
Can't use most commands with GitHub Enterprise, error message is misleading

### ISSUE THREAD

**Issue by duckpuppy** [github/hub#1544]
I have tried everything below with both 2.2.9 and 2.3.0-pre10

I have been using hub with a GHE instance at work for a very long time and have become very used to `git fork`, `git browse` and `git pull-request` as part of my workflow.  We have switched to a new instance (a centralized instance ran by another team), and those commands no longer work for me.  I can export the `GITHUB_HOST` variable and `git clone <username>/<repo>` works fine, so I know it's authenticating properly.  When I use any of the additional hub commands, though, all I get is

```
Aborted: the origin remote doesn't point to a GitHub repository.
```

I'm not sure how to troubleshoot further.  I've tried several things, and the repo does point to a GitHub repo because I just cloned it from there.  My basic test was

```
$ export GITHUB_HOST=<host>
$ git clone <org>/<repo>
$ cd <repo>
$ git fork
```

The clone succeeded just fine, but the fork command returned the above error.  There should be more information available to determine what the real problem is.

**Comment by mislav** [comment]
Have you followed the instructions outlined in "GitHub Enterprise" section of the man page? https://hub.github.com/hub.1.html

**Comment by sinisterstuf** [comment]
Regarding the recommended

    GITHUB_HOST=my.git.org git clone myproject

I ended up making an alias for that

    alias ghe='GITHUB_HOST=my.git.org hub'

and run the clone commands with `ghe` instead of `git` because it's too annoying to type out. Not sure if a better solution could be possible at all.

**Comment by generatz** [comment]
@sinisterstuf: Agreed, would be good to have an integrated solution.  Wondering if hub.host could be used.

**Comment by sinisterstuf** [comment]
To be honest I can't really see a way hub could guess which host you want based only on

```
git clone myproject
```

which is why I'm not sure there's going to be a much better solution for this. But as a GHE user I'd be really interested if someone has a solution!

**Comment by duckpuppy** [comment]
It's not so much about hub guessing from a `git clone`... that has been working by exporting the environment variable, and I never expected it to work without it.  I too have aliases for git to handle setting the `GITHUB_HOST` variable (two of them now, since I have to switch between the two).

The variable is needed for "initial" commands like `clone` or `remote add` if you're using the abbreviated GHE repo identifiers or new GHE specific commands like `fork`, but wasn't for commands that work on existing repos (particularly the GitHub specific ones like `pull-request` or `browse`) as the appropriate URLs are part of the remote - as long as your remotes are named correctly (`github`, `upstream`, etc).

The problem is that I couldn't get `hub pull-request` and the like to work with the new repo for some reason.

I "fixed" it, though I'm not sure what did it.  I had added a second 'hub.host' entry in my gitconfig for the new GHE instance.  Things worked for the old one, but not the new.

I removed the old one, which made `hub pull-request` etc. stop working for that host (as it should), but the new one started working.

I then later decided to try adding the old one back... and BOTH worked just fine. I have no idea what was different, other than the order of the entries in my gitconfig.
